skbio/metadata/_mixin.py

```diff
@@ -406,3 +407,114 @@ def has_positional_metadata(self):
         """
         return (self._positional_metadata is not None and
                 len(self.positional_metadata.columns) > 0)
+
+
+class IntervalMetadataMixin(metaclass=abc.ABCMeta):
+    def _interval_metadata_axis_len_(self):
+        ''''''
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def __init__(self, interval_metadata=None):
+        raise NotImplementedError
+
+    def _init_(self, interval_metadata=None):
+        if interval_metadata is None:
+            # Could use deleter but this is less overhead and needs to be fast.
+            self._interval_metadata = None
+        else:
+            # Use setter for validation and copy.
+            self.interval_metadata = interval_metadata
+
+    @property
+    @experimental(as_of="0.5.0-dev")
+    def interval_metadata(self):
+        '''``IntervalMetadata`` object containing info about interval features.
```

Comment:
Can you add a `Notes` section, similar to metadata/positional metadata? Something like:

```
Notes
-----
This property can be set and deleted. When setting new interval metadata, a
shallow copy of the `IntervalMetadata` object is made.
```

We'll need an `Examples` section too but will defer that until this class is hooked up to `Sequence` in another PR.